File: bot-api/src/main/java/gg/mineral/bot/api/inv/potion/Potion.java

package gg.mineral.bot.api.inv.potion;

import gg.mineral.bot.api.entity.effect.PotionEffect;

import java.util.List;

public interface Potion {
    /**
     * Returns whether this potion is a splash potion.
     *
     * @return true if the potion is a splash potion, false otherwise.
     */
    boolean isSplash();

    /**
     * Returns the list of potion effects.
     *
     * @return the list of potion effects
     */
    List<PotionEffect> getEffects();
}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/inv/potion/Potion.kt
package gg.mineral.bot.api.inv.potion

import gg.mineral.bot.api.entity.effect.PotionEffect

interface Potion {
    /**
     * Returns whether this potion is a splash potion.
     *
     * @return true if the potion is a splash potion, false otherwise.
     */
    val isSplash: Boolean

    /**
     * Returns the list of potion effects.
     *
     * @return the list of potion effects
     */
    val effects: List<PotionEffect>
}
